File: binary-number-with-alternating-bits/binary_number_with_alternating_bits.js
/**
 * @param {number} n
 * @return {boolean}
 */
var hasAlternatingBits = function(n) {
    var b = n.toString(2);
    var index = (b.indexOf("00") > b.indexOf("11")) ? b.indexOf("00") : b.indexOf("11");
    return index >= 0 ? false : true;
}

File: binary-number-with-alternating-bits/binary_number_with_alternating_bits.go
func hasAlternatingBits(n int) bool {
    remainder := n % 2
    n /= 2
    for n > 0 {
        new_remainder := n % 2
        if new_remainder == remainder {
            return false
        }
        n /= 2
        remainder = new_remainder
    }
    return true
}